src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === undefined && i === this.length - 1)
+      || (this[i] === null && i === this.length - 1)) {
+        result += '';
+      } else if (this[i] === undefined || this[i] === null) {
+        result += '' + (separator + '');
```

Comment:
Why do you need to add an empty string at the end? There is no sense to do this.